src/main/java/frc/robot/subsystems/Feeder.java

```diff
@@ -13,18 +14,33 @@
 import frc.robot.utils.diag.DiagColorSensor;
 import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
 
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
 public class Feeder extends SubsystemBase {
 
     private final WPI_TalonSRX feederMotor;
     private final I2C.Port i2cPort = I2C.Port.kMXP;
     private final ColorSensor colorSensor;
+    private final BlockingQueue<ColorMatchResult> colorReadingBuffer = new LinkedBlockingQueue<>();
```

Comment:
Blocking queue is not right here (if bound in size) since it will block the writing if full and the reading if empty.

Reply:
replaced with atomic integer